File: GenerateMd5/GenerateMd5.java
package Generate_MD5;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class GenerateMd5 {
    public static void main(String[] args) throws NoSuchAlgorithmException {
        System.out.println("Generate a MD5 hash");

        System.out.println("\n# # # SECURITY WARNING: This code is provided for achieve    # # #");
        System.out.println("# # # compatibility between different programming languages. # # #");
        System.out.println("# # # It is NOT SECURE - DO NOT USE THIS CODE ANY LONGER !   # # #");
        System.out.println("# # # The hash algorithm MD5 is BROKEN.                      # # #");
        System.out.println("# # # DO NOT USE THIS CODE IN PRODUCTION                     # # #\n");

        String plaintext = "The quick brown fox jumps over the lazy dog";
        System.out.println("plaintext:                " + plaintext);

        byte[] md5Value = calculateMd5(plaintext);
        System.out.println("md5Value (hex) length: " + md5Value.length + " data: " + bytesToHex(md5Value));
    }

    private static byte[] calculateMd5 (String string) throws NoSuchAlgorithmException {
        MessageDigest md = MessageDigest.getInstance("MD5");
        return md.digest(string.getBytes(StandardCharsets.UTF_8));
    }

    private static String bytesToHex(byte[] bytes) {
        StringBuffer result = new StringBuffer();
        for (byte b : bytes) result.append(Integer.toString((b & 0xff) + 0x100, 16).substring(1));
        return result.toString();
    }

}

File: GenerateMd5/GenerateMd5.cs
using System;
using System.Security.Cryptography;
using System.Text;

public class Program {
	public static void Main() {
		Console.WriteLine("Generate a MD5 hash");

    Console.WriteLine("\n# # # SECURITY WARNING: This code is provided for achieve    # # #");
    Console.WriteLine("# # # compatibility between different programming languages. # # #");
    Console.WriteLine("# # # It is NOT SECURE - DO NOT USE THIS CODE ANY LONGER !   # # #");
    Console.WriteLine("# # # The hash algorithm MD5 is BROKEN.                      # # #");
    Console.WriteLine("# # # DO NOT USE THIS CODE IN PRODUCTION                     # # #\n");
 
    string plaintext = "The quick brown fox jumps over the lazy dog";
    Console.WriteLine("plaintext:                " + plaintext);

    byte[] md5Value = CalculateMd5(plaintext);
    Console.WriteLine("md5Value (hex) length: " + md5Value.Length + " data: " + BytesToHex(md5Value));
	}

  static byte[] CalculateMd5(string input) {
    MD5 md5 = MD5.Create();
    return md5.ComputeHash(System.Text.Encoding.UTF8.GetBytes(input));
  }

  static string BytesToHex(byte[] ba) {
    StringBuilder hex = new StringBuilder(ba.Length * 2);
    foreach (byte b in ba)
      hex.AppendFormat("{0:x2}", b);
    return hex.ToString();
  }
}
